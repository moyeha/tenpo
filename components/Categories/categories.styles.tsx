import { Text } from "react-native";
import styled from "styled-components";
import theme from "../../theme";

const StyledText = styled(Text)({
  fontFamily: "Gotham-Black",
  fontSize: "18px",
  color: theme.colors.darkGray,
  fontWeight: "bold",
  textTransform: "uppercase",
});

export { StyledText };
