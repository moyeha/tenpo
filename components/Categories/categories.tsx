import React from "react";
import { useTranslation } from "react-i18next";
import { FlatList, View } from "react-native";
import { useAppSelector } from "../../hooks/useStore";
import Category from "../Category/category";
import { StyledText } from "./categories.styles";

const Categories = () => {
  const { t } = useTranslation();

  const categories = useAppSelector((state) => state?.categories);

  return (
    <View
      style={{
        marginTop: 20,
      }}
    >
      <StyledText>{t("categories-title")}</StyledText>
      <FlatList
        data={categories}
        style={{ marginTop: 10 }}
        horizontal
        renderItem={({ item }) => <Category {...item} />}
        keyExtractor={(item) => item?.id}
      />
    </View>
  );
};

export default Categories;
