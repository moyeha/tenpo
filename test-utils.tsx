import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react-native";
import type { RenderOptions } from "@testing-library/react-native";
import type { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore } from "./store";
import type { AppStore, RootState } from "./store";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { ThemeProvider } from "styled-components";
import theme from "./theme";

const initialMetrics = {
  frame: { height: 0, width: 0, x: 0, y: 0 },
  insets: { bottom: 0, left: 0, right: 0, top: 0 },
};

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  Element: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  const Wrapper = ({ children }: PropsWithChildren<{}>) => {
    return (
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <SafeAreaProvider initialMetrics={initialMetrics}>
            {children}
          </SafeAreaProvider>
        </ThemeProvider>
      </Provider>
    );
  };
  return { store, ...render(Element, { wrapper: Wrapper, ...renderOptions }) };
}
