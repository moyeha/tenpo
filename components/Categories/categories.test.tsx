import { screen } from "@testing-library/react-native";
import React from "react";
import Categories from "./categories";
import { renderWithProviders } from "../../test-utils";

describe("Categories", () => {
  test("Deberia estar el titulo Categorias", async () => {
    renderWithProviders(<Categories />);
    expect(screen.queryByText(/Categorias/i)).toMatchSnapshot();
  });
});
