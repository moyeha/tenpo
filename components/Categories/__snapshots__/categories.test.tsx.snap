// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Deberia estar el titulo Categorias 1`] = `null`;
